geomesa-hbase/geomesa-hbase-server/src/main/scala/org/locationtech/geomesa/hbase/server/common/CoprocessorScan.scala

```diff
@@ -123,11 +135,21 @@ trait CoprocessorScan extends StrictLogging {
     }
 
     private def continue(): Boolean = {
-      if (controller.isCanceled) {
+      if (yieldPartialResults) {
+        logger.trace(s"Stopping aggregator $aggregator at row ${ByteArrays.printable(aggregator.getLastScanned)} and" +
+          s"returning intermediate results.")
+        // This check makes covers the HBase Version Aggregator case
+        if (aggregator.getLastScanned != null && !aggregator.getLastScanned.isEmpty) {
+          results.setLastScanned(ByteString.copyFrom(aggregator.getLastScanned))
+        }
+        false
+      } else if (controller.isCanceled) {
```

Comment:
we should probably re-order the if checks so that this is first, then the timeout check (assuming we don't return the last row there), then this check, so we don't bother trying to continue on the client in those cases.

Reply:
Excellent point.  Doing it.

Reply:
Done.